test/test_smtlib.cpp

```diff
@@ -0,0 +1,534 @@
+
+#include "LogicBlock/SMTLibLogicBlock.hpp"
+#include "LogicBlock/SMTLibLogicModel.hpp"
+#include "LogicTerm/Logic.hpp"
+#include "LogicTerm/LogicTerm.hpp"
+
+#include "gmock/gmock.h"
+
+using namespace logicbase;
+
+class TestSMTLib: public testing::TestWithParam<logicbase::OpType> {
+protected:
+    void SetUp() override {
+    }
+};
+
+TEST(TestSMTLib, ConstructDestruct) {
+    const auto smtLibLogic = std::make_unique<smtliblogic::SMTLogicBlock>(false, std::cout);
+}
+
+TEST(TestSMTLib, TestPrint) {
+    const auto smtLibLogic = std::make_unique<smtliblogic::SMTLogicBlock>(false, std::cout);
+
+    const LogicTerm a = smtLibLogic->makeVariable("a", CType::BOOL);
+    const LogicTerm b = smtLibLogic->makeVariable("b", CType::INT);
+    const LogicTerm c = smtLibLogic->makeVariable("c", CType::REAL);
+    const LogicTerm d = smtLibLogic->makeVariable("d", CType::BITVECTOR, 8);
+
+    const LogicTerm t = a && (b == LogicTerm(1));
+    t.print(std::cout);
+    t.prettyPrint(std::cout);
+
+    const LogicTerm f = (c == LogicTerm(1.0)) && (d == LogicTerm(1, 8));
+    f.print(std::cout);
+    f.prettyPrint(std::cout);
+
+    smtLibLogic->assertFormula(t);
+    smtLibLogic->assertFormula(f);
+
+    smtLibLogic->setOutputLogic(smtliblogic::SMTLibLogic::QF_BV);
+    smtLibLogic->produceInstance();
+}
+
+TEST(TestSMTLib, TestBasicBooleanAnd) {
+    std::stringstream ss;
+
+    const auto smtLibLogic = std::make_unique<smtliblogic::SMTLogicBlock>(false, ss);
+
+    const std::string output = "; Generated by LogicBlock\n"
+                               "(set-option :produce-models true)\n"
+                               "(set-logic QF_UF)\n"
+                               "(declare-const p_0 Bool)\n"
+                               "(assert (and p_0 (not p_0)))\n"
+                               "(check-sat)\n"
+                               "(get-model)\n"
+                               "(exit)\n";
+
+    const LogicTerm p = smtLibLogic->makeVariable("p", CType::BOOL);
+    const LogicTerm t = p && !p;
+    smtLibLogic->assertFormula(t);
+    smtLibLogic->setOutputLogic(smtliblogic::SMTLibLogic::QF_UF);
+    smtLibLogic->produceInstance();
+
+    ASSERT_EQ(ss.str(), output);
+}
+
+TEST(TestSMTLib, TestBasicBooleanOrXor) {
+    std::stringstream ss;
+
+    const auto smtLibLogic = std::make_unique<smtliblogic::SMTLogicBlock>(false, ss);
+
+    const std::string output = "; Generated by LogicBlock\n"

```

Comment:
Isn't this missing an "s", I.e., shouldn't it be "LogicBlocks"?

One additional thing: shouldn't such a header also include some version information?

Reply:
Took a bit to figure out a somewhat clean solution for that, but I think taking the version info from cmake is the best course of action